pyicloud/base.py

```diff
@@ -93,22 +149,24 @@ def authenticate(self):
         subsequent logins will not cause additional e-mails from Apple.
         """
 
+        logger.info("Authenticating as %s", self.user['apple_id'])
+
         data = dict(self.user)
 
         # We authenticate every time, so "remember me" is not needed
         data.update({'extended_login': False})
 
-        req = self.session.post(
-            self._base_login_url,
-            params=self.params,
-            data=json.dumps(data)
-        )
-
-        resp = req.json() if req.ok else {}
-        if 'dsInfo' not in resp:
+        try:
+            req = self.session.post(
+                self._base_login_url,
+                params=self.params,
+                data=json.dumps(data)
+            )
+        except PyiCloudAPIResponseError, error:
```

Comment:
These days, one usually uses `as` rather than the old-style two-argument except block.  I *think* that's been an option since maybe 2.6, and it's [required for Python3.x](https://www.python.org/dev/peps/pep-3110/).